packages/flutter/test/material/slider_test.dart

```diff
@@ -5179,4 +5231,36 @@ void main() {
     await gesture.up();
     await tester.pumpAndSettle();
   });
+
+  testWidgets('secondaryTrackValue updates correctly', (WidgetTester tester) async {
+    final List<Offset?> log = <Offset?>[];
+    final LoggingRoundedRectSliderTrackShape loggingTrackShape = LoggingRoundedRectSliderTrackShape(
+      secondaryOffsetLog: log,
+    );
+    final ThemeData theme = ThemeData(sliderTheme: SliderThemeData(trackShape: loggingTrackShape));
+    Widget buildSlider(double? secondaryTrackValue) {
+      return MaterialApp(
+        theme: theme,
+        home: Material(
+          child: Center(
+            child: Slider(
+              value: 0,
+              secondaryTrackValue: secondaryTrackValue,
+              onChanged: (double value) {},
+            ),
+          ),
+        ),
+      );
+    }
+
+    await tester.pumpWidget(buildSlider(null));
+    await tester.pumpAndSettle();
+    expect(log.last, isNull);
+    await tester.pumpWidget(buildSlider(0.2));
+    await tester.pumpAndSettle();
+    expect(log.last, const Offset(174.4, 300.0));
+    await tester.pumpWidget(buildSlider(0.5));
+    await tester.pumpAndSettle();
+    expect(log.last, const Offset(400.0, 300.0));
```

Comment:
```suggestion
    await tester.pumpWidget(buildSlider(null));
    await tester.pumpAndSettle();
    expect(log.last, isNull);

    await tester.pumpWidget(buildSlider(0.2));
    await tester.pumpAndSettle();
    expect(log.last, const Offset(174.4, 300.0));

    await tester.pumpWidget(buildSlider(0.5));
    await tester.pumpAndSettle();
    expect(log.last, const Offset(400.0, 300.0));
```

Just for better readability